### PR TITLE
T5702: SNMP add interface-mib max-interfaces-number and prefix

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

- Allow to configure only required interface prefixes
```
set service snmp mib interface 'eth'
set service snmp mib interface 'bond'
```
```
 include_ifmib_iface_prefix eth bond

Sets the interface name prefixes to include in the IF-MIB data collection. 
For servers with a large number of interfaces (ppp, dummy, bridge, etc) the IF-MIB
processing will take a large chunk of CPU for ioctl calls. A set of space separated 
interface name prefixes will reduce the CPU load for IF-MIB processing. 
For example, configuring "include_ifmib_iface_prefix eth dummy lo" will include only
interfaces with these prefixes and ignore all others for IF-MIB processing.
```
- Allow to configure maximum interface number
```
set service snmp mib interface-max '25'
```
```
  ifmib_max_num_ifaces NUM

Sets the maximum number of interfaces included in IF-MIB data collection. 
For servers with a large number of interfaces (ppp, dummy, bridge, etc) the IF-MIB 
processing will take a large chunk of CPU for ioctl calls (on Linux). 
Setting a reasonable maximum for the CPU used will reduce the CPU load for IF-MIB processing.
 For example, configuring "ifmib_max_num_ifaces 500" will include only the first 500 
interfaces based on ifindex and ignore all others for IF-MIB processing.
```
Without those options we have high CPU utilization with many dynamic interfaces (ipoe/l2tp/pppoe/etc)
snmp

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5702

## Component(s) name

snmp
## How to test
VyOS configuration:
```
set service snmp community public client '127.0.0.1'
set service snmp mib interface-max '25'
set service snmp mib interface 'eth'
set service snmp mib interface 'bond'

```
snmpd config:
```
# Interface MIB limits
ifmib_max_num_ifaces 25
include_ifmib_iface_prefix eth bond

```

## Smoketest result
```
vyos@r4:~$ /usr/libexec/vyos/tests/smoke/cli/test_service_snmp.py
test_snmp_basic (__main__.TestSNMPService.test_snmp_basic) ... ok
test_snmpv3_md5 (__main__.TestSNMPService.test_snmpv3_md5) ... ok
test_snmpv3_sha (__main__.TestSNMPService.test_snmpv3_sha) ... ok

----------------------------------------------------------------------
Ran 3 tests in 22.142s

OK

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
Why not make the CLI more generic by using:

```
set service snmp mib interface-max '25'
set service snmp mib interface 'eth'
set service snmp mib interface 'bond'
```

?

**Comment by sever-sever** [comment]
> Why not make the CLI more generic by using:
> 
> ```
> set service snmp mib interface interfaces-max '25'
> set service snmp mib interface 'eth'
> set service snmp mib interface 'bond'
> ```
> 
> ?

How to separate “interface interfaces-max” and “interface eth”?
Will interface be a “tagNode”?

**Comment by c-po** [comment]
> > Why not make the CLI more generic by using:
> > ```
> > set service snmp mib interface interfaces-max '25'
> > set service snmp mib interface 'eth'
> > set service snmp mib interface 'bond'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ?
> 
> How to separate “interface interfaces-max” and “interface eth”? Will interface be a “tagNode”?

Sorry, typo - I have adjusted m initial comment

**Comment by sever-sever** [comment]
> > > Why not make the CLI more generic by using:
> > > ```
> > > set service snmp mib interface interfaces-max '25'
> > > set service snmp mib interface 'eth'
> > > set service snmp mib interface 'bond'
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ?
> > 
> > 
> > How to separate “interface interfaces-max” and “interface eth”? Will interface be a “tagNode”?
> 
> Sorry, typo - I have adjusted m initial comment

Done!
```
set service snmp community public client '127.0.0.1'
set service snmp mib interface 'eth'
set service snmp mib interface 'bond'
set service snmp mib interface-max '25'
```

File: interface-definitions/snmp.xml.in [hunk 1]
```diff
@@ -79,6 +79,39 @@
...
+                  <constraint>
+                    <validator name="numeric" argument="--range 1-4294967295"/>
+                  </constraint>
+                </properties>
+              </leafNode>
+              <leafNode name="interface">
+                <properties>
+                  <help>Sets the interface name prefix to include in the IF-MIB data collection</help>
+                  <valueHelp>
+                    <format>txt</format>
```
Comment (c-po) [inline interface-definitions/snmp.xml.in]: Should we probably include proper `<valueHelp>` nodes for the interface types?

Like eth,br,bond.wireguard,vxlan and so on? So the users know only the text part is required and nothing like an asterisk as glob pattern?

**Review by c-po — APPROVED** [review]

File: interface-definitions/snmp.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/snmp.xml.in]: @c-po Done!

**Comment by sever-sever** [comment]
@MergifyIo backport sagitta

**Pull request merged**